aviary/mission/gasp_based/phases/time_integration_phases.py

```diff
@@ -504,9 +498,14 @@ def __init__(
         )
 
         self.phase_name = phase_name
-        self.add_trigger(Dynamic.Mission.MASS, 'mass_trigger.mass_trigger', units='lbm')
-        self.add_trigger(Dynamic.Mission.DISTANCE, "distance_trigger",
-                         units=self.distance_trigger_units)
+        self.add_trigger(Dynamic.Mission.DISTANCE, "distance_trigger")
+        from aviary.utils.test_utils.check_om_version import CheckForOMSubmodelFix
+        if CheckForOMSubmodelFix():
+            # using the value of the SGMCruise attribute
+            self.add_trigger(Dynamic.Mission.MASS, 'mass_trigger')
+        else:
+            # using the value of the FlightPathODE mass_trigger subsystem
+            self.add_trigger(Dynamic.Mission.MASS, 'mass_trigger.mass_trigger')
```

Comment:
Part 2 of 2 for this particular comment:
If you changed that `promote_outputs` that I referenced (and anywhere else as needed) could you remove this logic here and just use `mass_trigger` regardless? (not `mass_trigger.mass_trigger`).

Reply:
depending on whether or not you have the submodel fix, the phase has an attribute called `mass_trigger` that takes priority over connections (see `evaluate_trigger()` in time_integration_base_classes.py).
and again, this can be removed once the submodel fix is in a tagged release of OM